src/mpid/ch4/generic/mpidig_send.h

```diff
@@ -14,21 +14,62 @@
 
 #include "ch4_impl.h"
 
+#define MPIDIG_AM_SEND_HDR_SIZE  sizeof(MPIDIG_hdr_t)
```

Comment:
Will be available for other message types? Rendezvous shouldn't be limited to pt2pt operations.

Reply:
My idea is to have specific protocol for each specific operation. So a rendezvous pt2pt will be a distinct protocol from, say a put rendezvous protocol. In fact, even within pt2pt send, we may have different rndz protocols, e.g. one send lmt, one send multiple eager segments, one sets up rdma, etc., and each will be distinct from each other and each have independent headers.

That said, we still can reduce code-duplication by factoring common routines to an inline header file. But conceptually, they are distinct and can be modified locally in all aspects without interfering other similar protocols.

Reply:
I have to be honest I thought ch3 sent more message types over rndv protocol, but looking back I was wrong. This is probably fine to start here and add more as we go.